include/seastar/http/reply.hh

```diff
@@ -228,6 +228,10 @@ private:
 
 std::ostream& operator<<(std::ostream& os, reply::status_type st);
 
+namespace status_strings {
+const sstring& to_string(reply::status_type status);
```

Comment:
instead of exposing `to_string()`, i'd suggest add a `fmt::formatter<>` for it. or if we has no strong interests in the compatibility of fmtlib v9, could just go ahead and implement `format_as()`. so we can kill two birds with one stone:

- expose this helper to a wider audience like `unexpected_status_error` 's constructor
- and include fmtlib in the audience